OpenHardwareMonitor/HardwareMonitor.cs

```diff
@@ -35,38 +35,28 @@ public void Where(string property, string value, string op="=")
             this.statements.Add(string.Format("{0}{1}{2}", property, op, value));
         }
 
-        public ManagementObjectCollection Get()
+        public ManagementObject GetAt(int index)
         {
-            string query = this.ToString();
-            ManagementScope scope = new ManagementScope(this.ns);
-            scope.Connect();
-            using (ManagementObjectSearcher mos = new ManagementObjectSearcher(scope, new ObjectQuery(query)))
-            {
-                return mos.Get();
-            }
-        }
-
-        public ManagementObject GetAt(int i)
-        {
-            using(ManagementObjectCollection moc = this.Get())
-            {
-                return getObjectAt(moc, i);
-            }
-        }
-
-        internal ManagementObject getObjectAt(ManagementObjectCollection moc, int index)
-        {
-            int i = 0;
-            foreach (ManagementObject m in moc)
-            {
-                if (index == i)
+            using (ManagementObjectSearcher mos = new ManagementObjectSearcher(new ManagementScope(this.ns), new ObjectQuery(this.ToString())))
+                using (ManagementObjectCollection moc = mos.Get())
                 {
-                    return m;
+                    int i = 0;
+                    foreach (ManagementObject m in moc)
+                    {
+                        try
+                        {
+                            if (index == i)
+                                return m;
```

Comment:
Is it save to call `Dispose` on the returned `ManagementObject` m?

Reply:
Good question. Whole try/finally/dispose is probably redundant here because it is wrapped in using at https://github.com/mnaiman/Rainmeter-HardwareMonitor/blob/1b3051eaad35502a635646aaa3b61bc949ab4998/OpenHardwareMonitor/HardwareMonitor.cs#L128
But it is working ok. GetPropertyValue is probably not using managed resources that are freed before.